Modules/_ctypes/callproc.c

```diff
@@ -1992,7 +1999,51 @@ buffer_info(PyObject *self, PyObject *arg)
     return Py_BuildValue("siN", dict->format, dict->ndim, shape);
 }
 
+#ifndef MS_WIN32
+
+#ifdef HAVE_DL_ITERATE_PHDR
+static int interp_cb(struct dl_phdr_info *info, size_t size, void *data)
+{
+    const char **ps = data;
```

Comment:
What's this for? You do not use `ps` until the loop body, and there it's assigned a new value for each iteration.